recipes/fpgen/all/conanfile.py

```diff
@@ -1,94 +1,74 @@
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import copy, get, replace_in_file
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.52.0"
 
 
 class FpgenConan(ConanFile):
     name = "fpgen"
-    description = " Functional programming in C++ using C++20 coroutines."
+    description = "Functional programming in C++ using C++20 coroutines."
     license = ["MPL2"]
-    topics = (
-        "generators",
-        "coroutines",
-        "c++20",
-        "header-only",
-        "functional-programming",
-        "functional",
-    )
-    homepage = "https://github.com/jay-tux/fpgen/"
     url = "https://github.com/conan-io/conan-center-index"
-    settings = "arch", "os", "compiler", "build_type"
-    no_copy_source = True
+    homepage = "https://github.com/jay-tux/fpgen/"
+    topics = ("generators", "coroutines", "c++20", "header-only", "functional-programming", "functional")
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
 
     @property
     def _min_cppstd(self):
-        return "20"
+        return 20
 
     @property
     def _compilers_minimum_version(self):
         return {
             "gcc": "11",
             "clang": "13",
+            "apple-clang": "13.1",
+            "Visual Studio": "16",
+            "msvc": "192",
         }
 
-    def package_id(self):
-        self.info.header_only()
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
-    def source(self):
-        tools.get(
-            **self.conan_data["sources"][self.version],
-            destination=self._source_subfolder,
-            strip_root=True,
-        )
+    def package_id(self):
+        self.info.clear()
 
     def validate(self):
-        if self.settings.compiler == "clang" and "clang" not in str(self.version):
-            raise ConanInvalidConfiguration(
-                f"Use '{self.version}-clang' for Clang support."
-            )
-
-        if (
-            self.settings.compiler == "clang"
-            and not self.settings.compiler.libcxx == "libc++"
-        ):
-            raise ConanInvalidConfiguration(
-                f"Use 'compiler.libcxx=libc++' for {self.name} on Clang."
-            )
+        if self.settings.compiler == "clang" and not self.settings.compiler.libcxx == "libc++":
+            raise ConanInvalidConfiguration(f"Use 'compiler.libcxx=libc++' with Clang for {self.name}.")
 
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._min_cppstd)
-
-        def lazy_lt_semver(v1, v2):
-            lv1 = [int(v) for v in v1.split(".")]
-            lv2 = [int(v) for v in v2.split(".")]
-            min_length = min(len(lv1), len(lv2))
-            return lv1[:min_length] < lv2[:min_length]
```

Comment:
by removing this logic, you prevent to consume this library with apple-clang 13

Reply:
Thanks! Opened a PR: #19245